artifactory/services/utils/tests/catalog/server.go

```diff
@@ -0,0 +1,78 @@
+package catalog
+
+import (
+	"fmt"
+	"net/http"
```

Comment:
why not use "net/http/httptest" and move this to respective *_test.go file?

Reply:
I was following the same impl used in: `artifactory/services/utils/tests/xray`

Reply:
may not be part of this PR, but this can be improved by using net/http/httptest